src/openrct2/ride/Vehicle.cpp

```diff
@@ -7288,16 +7288,15 @@ void Vehicle::UpdateAdditionalAnimation()
     switch (vehicleEntry->animation)
     {
         case VEHICLE_ENTRY_ANIMATION_MINITURE_RAILWAY_LOCOMOTIVE: // loc_6D652B
-            *curVar_C8 += _vehicleVelocityF64E08;
-            al = (*curVar_C8 >> 20) & 3;
-            if (animation_frame != al)
-            {
-                ah = al;
-                al = animation_frame;
-                animation_frame = ah;
-                al &= 0x02;
-                ah &= 0x02;
-                if (al != ah)
+            *curAnimation_state += _vehicleVelocityF64E08;
+            target_frame = (*curAnimation_state >> 20) & 3;
+            if (animation_frame != target_frame)
+            {
+                steam_frame = animation_frame;
+                animation_frame = target_frame;
```

Comment:
Are these two lines the refactor you’re talking about?

Reply:
yes